src/lib/storage/table.cpp

```diff
@@ -18,56 +18,80 @@
 namespace opossum {
 
 Table::Table(const ChunkOffset target_chunk_size) {
-  // Implementation goes here
+  _target_chunk_size = target_chunk_size;
+  _chunks.emplace_back(Chunk());
 }
 
 void Table::add_column(const std::string& name, const std::string& type) {
-  // Implementation goes here
+  DebugAssert(_chunks.size() == 1, "Invalid amount of chunks");
+  DebugAssert(row_count() == 0, "Too many rows");
+
+  _add_segment(type);
+
+  _column_names.emplace_back(name);
+  _column_types.emplace_back(type);
 }
 
-void Table::append(const std::vector<AllTypeVariant>& values) {
-  // Implementation goes here
+void Table::_add_segment(const std::string& type) {
+  resolve_data_type(type, [&](const auto data_type_t) {
+    using ColumnDataType = typename decltype(data_type_t)::type;
+    const auto value_segment = std::make_shared<ValueSegment<ColumnDataType>>();
+    _chunks.back().add_segment(value_segment);
+  });
 }
 
-ColumnCount Table::column_count() const {
-  // Implementation goes here
-  return ColumnCount{0};
+void Table::_create_new_chunk() {
+  _chunks.emplace_back(Chunk());
+  for (size_t column_id = 0; column_id < _column_names.size(); ++column_id) {
+    auto type = _column_types.at(column_id);
+    _add_segment(type);
+  }
 }
 
-uint64_t Table::row_count() const {
-  // Implementation goes here
-  return 0;
+void Table::append(const std::vector<AllTypeVariant>& values) {
+  DebugAssert(_chunks.size() > 0, "No chunks present");
+
+  if (_chunks.back().size() >= _target_chunk_size) {
+    _create_new_chunk();
+  }
+
+  _chunks.back().append(values);
 }
 
-ChunkID Table::chunk_count() const {
-  // Implementation goes here
-  return ChunkID{0};
+ColumnCount Table::column_count() const { return _chunks.at(0).column_count(); }
```

Comment:
Currently, chunks can not be deleted. The `operator[]` should be sufficient and could improve performance because the bounds don't need to be checked.
```suggestion
ColumnCount Table::column_count() const { return _chunks[0].column_count(); }
```